src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = (numberToAdd) => {
+    if (numberToAdd != null) {
+      amount += numberToAdd;
+
+      return adder;
+    } else {
```

Comment:
redundant else.
remove wrapping rest code in else.
in if block already exists return.